lib/registry.go

```diff
@@ -33,45 +33,47 @@ func (RegistryClientMethods) CoreRequestsName() string { return "registry" }
 type RegistryProfile = registry.Profile
 
 // CreateProfile creates a profile
-func (m RegistryClientMethods) CreateProfile(p *RegistryProfile, ok *bool) (err error) {
-	if m.inst.rpc != nil {
-		return checkRPCError(m.inst.rpc.Call("RegistryClientMethods.CreateProfile", p, ok))
+func (m RegistryClientMethods) CreateProfile(ctx context.Context, p *RegistryProfile) (bool, error) {
```

Comment:
Can we keep this as only returning the error? In my opinion, it's an anti-pattern to return both a `bool` and `error` if the bool's only purpose is to be equivalent to `err == nil`. Rather, it's perfectly fine to just use the error's presence or absence to determine success or failure. Especially since all the callers in this PR are ignoring the `bool` part of the return; a sure sign that it's not needed.

Reply:
The only purpose is to conform to the original dispatch method format. Has that changed? Looking at the code we still require 2 output params.

Would love to get rid of it though

Reply:
Ah, I see, yes sorry for the confusion. The original dispatch required 2 return values, but it's valid now to return 1-3, with 1 being just an error.